nandtool/nandtool.c

```diff
@@ -177,6 +177,91 @@ static int nandtool_check(void)
 }
 
 
+static int nandtool_dump(const char *outpath, unsigned int start, unsigned int nblocks, int oob)
+{
+	flashsrv_info_t *info;
+	int ret;
+	size_t blocksz, partsz, pagesz;
+	size_t addr, endaddr;
+	uint8_t *buf;
+	int outfd;
+
+	outfd = open(outpath, O_RDWR | O_CREAT);
+	if (outfd < 0) {
+		fprintf(stderr, "nandtool: Fail to open file: %s\n", outpath);
+		return -EINVAL;
+	}
+
+	info = flashmng_info(nandtool_common.oid);
+	if (info == NULL) {
+		close(outfd);
+		fprintf(stderr, "nandtool: Fail to retrieve partition info\n");
+		return -EFAULT;
+	}
+
+	if (!oob) {
+		blocksz = info->erasesz;
+		partsz = info->size;
+		pagesz = info->writesz;
+	}
+	else {
+		pagesz = info->metasz + info->writesz;
+		blocksz = (info->erasesz / info->writesz) * pagesz;
+		partsz = (info->size / info->erasesz) * blocksz;
+	}
+
+	addr = start * blocksz;
+
+	if (!oob) {
+		if (lseek(nandtool_common.fd, addr, SEEK_SET) < 0) {
+			fprintf(stderr, "nandtool: lseek failed\n");
```

Comment:
use `perror` or `strerror` for more meaningful error (applies also  to `open` error case).

Also: be consistent about operations order on every code path (sometimes you first close the file, than print the error, sometimes it's the other way around)